pkg/controllers/apps/resource/resource.go

```diff
@@ -69,25 +102,43 @@ func NewController(apiResource *metav1.APIResource, clusternetClient *versioned.
 		AddFunc: func(obj interface{}) {
 			resource := obj.(*unstructured.Unstructured)
 			val, ok := resource.GetAnnotations()[known.ObjectOwnedByDescriptionAnnotation]
-			if ok {
+			resAttrs, err := getResourceAttrs(val, ObjectAdd)
+			if ok && err == nil {
 				klog.V(4).Infof("adding %s %q", c.name, klog.KObj(resource))
-				c.enqueue(val)
+				c.enqueue(resAttrs)
 			}
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
-			resource := oldObj.(*unstructured.Unstructured)
-			val, ok := resource.GetAnnotations()[known.ObjectOwnedByDescriptionAnnotation]
-			if ok {
-				klog.V(4).Infof("updating %s %q", c.name, klog.KObj(resource))
-				c.enqueue(val)
+			oldresource := oldObj.(*unstructured.Unstructured)
+			newresource := newObj.(*unstructured.Unstructured)
+			oldstatus, _, _ := unstructured.NestedMap(oldresource.Object, "status")
+			newstatus, _, _ := unstructured.NestedMap(newresource.Object, "status")
+			oldspec, _, _ := unstructured.NestedMap(oldresource.Object, "spec")
+			newspec, _, _ := unstructured.NestedMap(newresource.Object, "spec")
+			val, ok := oldresource.GetAnnotations()[known.ObjectOwnedByDescriptionAnnotation]
+
+			var resAttrs *ResourceAttrs
+			var err error
+			if !reflect.DeepEqual(oldspec, newspec) {
```

Comment:
```
if !reflect.DeepEqual(oldstatus, newstatus) {
     resAttrs, err = getResourceAttrs(val, ObjectUpdateStatus)
} else {
      resAttrs, err = getResourceAttrs(val, ObjectUpdateOthers)
}

Reply:
> ```
> if !reflect.DeepEqual(oldstatus, newstatus) {
>      resAttrs, err = getResourceAttrs(val, ObjectUpdateStatus)
> } else {
>       resAttrs, err = getResourceAttrs(val, ObjectUpdateOthers)
> }
> ```
If spec and status change at the same time, the logic here needs to process spec first.